cmd/publish/main.go

```diff
@@ -46,17 +46,22 @@ func New(ctx *PublishCommandContext) *cobra.Command {
 			}
 			return nil
 		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			outputFlag, _ := cmd.Flags().GetString("output")
```

Comment:
👑

Reply:
Hi @noaabarki, first of all Congratulations for 2022 People's Choice Award for [Kubernetes](https://opensource.com/tags/kubernetes) :tada:.

I agree with your point :100:! As per my observation refactoring will require a lot of changes and can go out of scope of this PR. Example: I was going through the `cmd/root.go` file and realised we can refactor code for all of the commands. Then we may need to update test cases as well. That's a lot of work!

My suggestion is to create a new issue and work on it.  Looking forward to hearing from you!

Reply:
Sounds wonderful! Do you want to take the lead on this issue?

Reply:
Hi @noaabarki , yeah sure I can take it up.